vector/src/main/java/org/apache/arrow/vector/DecimalVector.java

```diff
@@ -564,8 +564,7 @@ private class TransferImpl implements TransferPair {
     DecimalVector to;
 
     public TransferImpl(String ref, BufferAllocator allocator) {
-      to =
-          new DecimalVector(ref, allocator, DecimalVector.this.precision, DecimalVector.this.scale);
+      to = new DecimalVector(ref, DecimalVector.this.field.getFieldType(), allocator);
```

Comment:
Same here. AFAIR, `field` and `getFieldType()` can be `null` (even if it should not happen, nothing prevent that in the code).